prototyping/api_tables.ipynb

```diff
@@ -0,0 +1,613 @@
+    "    filtered = filter_raw_card(raw_card)\n",
+    "    card = get_card_dataframe(filtered)\n",
+    "\n",
+    "    # Rename 'titleId' to 'card_id' and promote it to index\n",
+    "    for df in [filtered, card]:\n",
+    "        df.set_index('titleId', inplace=True)\n",
+    "        df.index.name = 'card_id'\n",
+    "\n",
+    "    card_type = get_card_type(filtered)\n",
+    "    card_subtype = get_card_subtype(filtered)\n",
+    "    card_cost = get_card_cost(filtered)\n",
+    "    card_ability = get_card_ability(filtered)\n",
+    "\n",
+    "    return card, card_type, card_subtype, card_cost, card_ability\n",
+    "\n",
+    "card, card_type, card_subtype, card_cost, card_ability = get_card_information()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Analytics\n",
+    "\n",
+    "The main function in the following block is `get_analytics()`. It gets data from `raw_analytics`. Normalization results in the following data frames with respective schema:\n",
+    "\n",
+    "- `analytics_games`: Defined by `card_id`, `timestamp`, `tier` and `copies`\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `timestamp` (Datetime PK): datetime from data aquisition\n",
+    "    - `tier` (String PK): game tier, from bronze to platinum\n",
+    "    - `copies` (Integer PK): number of copies, between 1 and 4\n",
+    "    - `games`(Integer): number of games\n",
+    "- `analytics_wins`: Defined by `card_id`, `timestamp` and `tier`\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `timestamp` (Datetime PK): datetime from data aquisition\n",
+    "    - `tier` (String PK): game tier, from bronze to platinum\n",
+    "    - `wins` (Integer): number of games resulting in victory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def filter_raw_analytics():\n",
+    "    # Reset index and rename tiers\n",
+    "    df = (raw_analytics.reset_index()\n",
+    "                       .rename(columns={'index':'raw'}))\n",
+    "    \n",
+    "    # Split raw column into multiple\n",
+    "    df[['card_id', 'archetype_id', 'tier']] = df.raw.str.split('.', expand=True)\n",
+    "\n",
+    "    # Record only consolidated data by archetype and change index to titleId\n",
+    "    df = (df[df.archetype_id == 'ALL']\n",
+    "            .drop(['raw', 'archetype_id'], axis=1)\n",
+    "            .set_index('card_id'))\n",
+    "    \n",
+    "    # Replace abbreviations with tier full names\n",
+    "    df.replace({'b': 'Bronze',\n",
+    "                's': 'Silver',\n",
+    "                'g': 'Gold',\n",
+    "                'p': 'Platinum'},\n",
+    "               inplace = True)\n",
+    "    \n",
+    "    # Time stamp\n",
+    "    df['timestamp'] = pd.Timestamp.today()\n",
```

Comment:
Will this column be saved on the database? If yes, the database can do that for you automatically: https://alvinalexander.com/android/sqlite-default-datetime-field-current-time-now/

Reply:
Yeah, makes sense...

I thought it would be best to record the timestamp before dividing the table due to normalization. But now I'm thinking the process will be something like this:
- Record the `analytics_wins` table on database;
- Leave `id` and `timestamp` to be defined by the DBMS
- Query the `analytics_wins.id` for the most recent hour (updates to the API are done hourly)
- Using these queried ids, fill `games_id` in the `analytics_games` table in order to avoid repeating the trio [card_id, timestamp, tier]
- Record the  `analytics_games` table on database.

What do you think?
For now, I simply removed the timestamp column.